src/components/Modal/Modal.stories.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState } from 'react';
+import { Meta, Story } from '@storybook/react/types-6-0';
+import { action } from '@storybook/addon-actions';
+
+import Modal from './Modal';
+import { ModalProps } from './Modal.types';
+import { ModalWidthSizes } from './Modal.enums';
+import { H2 } from '../typography';
+import { Inline, Padbox } from '../layout';
+import { Button, ButtonEnums } from '../Button';
+import { generateControl } from '../../utils/tests/storybook';
+import { SpaceSizes } from '../../index';
+
+export default {
+  title: 'components/modals/Modal',
+  component: Modal,
+  argTypes: {
+    width: {
+      ...generateControl('select', ModalWidthSizes),
+    },
+    footer: {
+      control: { disable: true },
+    },
+  },
+} as Meta;
+
+const Footer = () => (
+  <Inline gap={SpaceSizes.md} justify="end">
+    <Button size={SpaceSizes.lg} variant="outline">
+      Cancel
+    </Button>
+    <Button size={SpaceSizes.lg}>Confirm</Button>
+  </Inline>
+);
+
+export const Playground: Story<ModalProps> = (args) => {
+  const [isShown, setIsShown] = useState(false);
+
+  return (
+    <Padbox paddingSize={SpaceSizes.xxl}>
+      <Inline justify="center">
+        <Button
+          size={ButtonEnums.ButtonSizes.lg}
+          onClick={() => setIsShown(!isShown)}
+        >
+          Open modal
+        </Button>
+        {isShown && (
+          <Modal
+            footer={<Footer />}
+            title="Title"
+            onClose={() => setIsShown(false)}
+            {...args}
+          >
+            <H2 style={{ margin: 0 }}>This is a placeholder</H2>
+            You can replace this with a local component. You can also just
+            override the text, but keep in mind that you might lose content that
+            way in case we happend to change the underlying component. Lorem
+            ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
+            tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim
+            veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex
+            ea commodo consequat.
+          </Modal>
+        )}
+      </Inline>
+    </Padbox>
+  );
+};
+
```

Comment:
please disable visual tests for Playground story (it contains only the trigger button :) )
```js
Playground.parameters = {
  screenshot: { skip: true },
};
```